python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -14,98 +14,202 @@
-                                       cosmologicalDistanceModulus = None, specFileMap=None,
-                                       indices=None):
+    #an object carrying around photometric parameters like readnoise, effective area, plate scale, etc.
+    #defaults to LSST values
+    photParams = PhotometricParameters()
 
+    def calculateMagnitudeUncertainty(self, magnitudes, bandpassDict, obs_metadata=None):
         """
-        Calculate the magnitudes for different components (disk, bulge, agn, etc) of galaxies.
-        This method is designed to be used such that you feed it all of the disk Seds from your data
-        base and it returns the associated magnitudes.  Then you feed it all of the bulge Seds, etc.
+        Calculate the uncertainty in magnitudes using the model from equation (5) of arXiv:0805.2366
 
-        @param [in] objectID is the name of the galaxies (the whole galaxies)
+        @param [in] magnitudes is a numpy array containing the object magnitudes.  Every row corresponds to
+        a bandpass, which is to say that magnitudes[i][j] is the magnitude of the jth object in the
+        bandpass characterized by self.bandpassDict.values()[i]
 
-        @param [in] componentNames gives the name of the SED filenames
+        @param [in] bandpassDict is a BandpassDict characterizing the bandpasses being used
 
-        @param [in] magNorm is the normalizing magnitude
+        @param [in] obs_metadata is the metadata of this observation (mostly desired because
+        it will contain information about m5, the magnitude at which objects are detected
+        at the 5-sigma limit in each bandpass)
 
-        @param [in] internalAv is the internal Av extinction
+        @param [out] a 1-dimensional numpy array containing the magntiude uncertainty
+        corresponding to the magnitudes passed into this method.
+        """
 
-        @param [in] redshift is pretty self-explanatory
+        if obs_metadata is None:
+            raise RuntimeError("Need to pass an ObservationMetaData into calculatePhotometricUncertainty")
 
-        @param [in] cosmologicalDistanceModulus is the effective distance modulus due to cosmological
-        expansion (if that has not already been accounted for in magNorm).  This is optional.
+        if magnitudes.shape[0] != len(bandpassDict):
+            raise RuntimeError("Passed %d magnitudes to " % magnitudes.shape[0] + \
+                                " PhotometryBase.calculatePhotometricUncertainty; " + \
+                                "needed %d " % len(bandpassDict))
 
-        @param [in] specFileMap is a mapping between the filenames in diskNames, bulgeNames, and agnNames
-        and the absolute locations of the corresponding files.  It is an instantiation of the class defined
-        in
+        #if we have not run this method before, calculate and cache the m5 and gamma parameter
+        #values (see eqn 5 of arXiv:0805.2366) for future use
+        m5Defaults = None
```

Comment:
Can we make this a keyword argument to the method with default None? Otherwise, maybe we can have a more LSST specific name rather than having something that sounds more general and always assigning LSST defaults to it.
I suppose that is that plan for later anyway.